frontend/src/app/modules/time-series/services/line-chart.service.ts

```diff
@@ -245,6 +248,35 @@ export class LineChartService {
       .nice();
   }
 
+  private getLegendHeight(): number {
+    let maximumLabelWidth: number = 1;
+    let labels = Object.keys(this.legendDataMap);
+
+    d3Select('g#time-series-chart-legend')
+      .append('g')
+      .attr('id', 'renderToCalculateMaxWidth')
+      .selectAll('.renderToCalculateMaxWidth')
+      .data(labels)
+      .enter()
+      .append('text')
+      .attr('class', 'legend-text')
+      .text(datum => this.legendDataMap[datum].text)
+      .each((datum, index, groups) => {
+        Array.from(groups).forEach((text) => {
+          if (text) {
+            maximumLabelWidth = Math.max(maximumLabelWidth, text.clientWidth)
```

Comment:
Since Firefox does not support `clientWidth` for `svg` elements we should use this instead:
```suggestion
            maximumLabelWidth = Math.max(maximumLabelWidth, text.getBoundingClientRect().width)
```